MerlinAU.sh

```diff
@@ -5447,19 +5476,21 @@ _RunOfflineUpdateNow_()
         then
             if "$isGNUtonFW"
             then
-                # Extract the filename from the path
-                original_filename=$(basename "$selected_file")
-                # Sanitize filename by removing problematic characters (if necessary)
-                sanitized_filename=$(echo "$original_filename" | sed 's/[^a-zA-Z0-9._-]//g')
-                # Extract the file extension
+                # Extract the filename from the path #
+                original_filename="$(basename "$selected_file")"
+                # Sanitize filename by removing problematic characters (if necessary) #
+                sanitized_filename="$(echo "$original_filename" | sed 's/[^a-zA-Z0-9._-]//g')"
+                # Extract the file extension #
                 extension="${sanitized_filename##*.}"
                 FW_DL_FPATH="${FW_ZIP_DIR}/${FW_FileName}.${extension}"
                 _GnutonBuildSelection_
                 set -- $(_GetLatestFWUpdateVersionFromGithub_ "$FW_GITURL_RELEASE" "$firmware_choice")
+                retCode="$?"
             else
                 set -- $(_GetLatestFWUpdateVersionFromWebsite_ "$FW_SFURL_RELEASE")
+                retCode="$?"
             fi
-            if [ $? -eq 0 ] && [ $# -eq 2 ] && \
+            if [ "$retCode" -eq 0 ] && [ $# -eq 2 ] && \
```

Comment:
The code lines **5488**, **5491** & **5493** are examples where code was added/modified due to non-compliance with coding standards & best practices. Essentially, you always want to grab the return code of a function right after the function call **without** any other intervening code.

Reply:
Ah yes, I see/understand. Thanks for the example!